### PR TITLE
Feat: EnableAddon supports runtime cluster

### PR DESCRIPTION
**Pull request by hongchaodeng**
If use runtime cluster mode, the definitions will be applied to control plane k8s directly, not included in the Application object.

### CODE REVIEW

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -362,26 +363,60 @@ func RenderApplication(addon *types.Addon, args map[string]string) (*v1beta1.App
...
+}
+
+func isDeployToRuntimeOnly(addon *types.Addon) bool {
+	if addon.DeployTo == nil {
+		return false
+	}
+	if !addon.DeployTo.RuntimeCluster {
+		return false
+	}
+	return true
```
Comment (chivalryq) [inline pkg/addon/addon.go:409]: just `return addon.DeployTo.RuntimeCluster`

File: pkg/addon/addon.go [hunk 2]
```diff
@@ -362,26 +363,60 @@ func RenderApplication(addon *types.Addon, args map[string]string) (*v1beta1.App
...
+			app.Spec.Components = append(app.Spec.Components, *comp)
 		}
-		app.Spec.Components = append(app.Spec.Components, *comp)
 	}
 
-	return app, nil
+	return app, defObjs, nil
+}
+
+func isDeployToRuntimeOnly(addon *types.Addon) bool {
```
Comment (chivalryq) [inline pkg/addon/addon.go:402]: Why is `isDeployToRuntimeOnly` , why can't flux be installed both runtime and control cluster?

**Review by chivalryq — CHANGES_REQUESTED** [review]
I just feel a bit confused

File: pkg/addon/addon.go (hunk 2 above)
Comment (hongchaodeng) [reply pkg/addon/addon.go:402]: Yes. It will..
The runtime clusters include the control plane.

**Pull request merged**